chainer/functions/pooling/average_pooling_nd.py

```diff
@@ -0,0 +1,155 @@
+            average_pooling_nd_kernel.AveragePoolingNDKernelForward.generate(
+                self.ndim)
+        cuda.elementwise(in_params, out_params, operation, name)(
+            x[0].reduced_view(),
+            *(dims + ys + self.ksize + self.stride + self.pad + (coeff, y)))
+
+        return y,
+
+    def backward_cpu(self, x, gy):
+        dims = x[0].shape[2:]
+        outs = gy[0].shape[2:]
+        colon = slice(None, None, None)
+        gy_index = (colon, colon) + (None,) * len(dims)
+        gcol_reps = (1, 1) + self.ksize + (1,) * len(outs)
+        gcol = numpy.tile(gy[0][gy_index], gcol_reps)
+        gx = conv_nd.col2im_nd_cpu(gcol, self.stride, self.pad, dims)
+        gx /= functools.reduce(operator.mul, self.ksize)
+        return gx,
+
+    def backward_gpu(self, x, gy):
+        if (cuda.cudnn_enabled and self.use_cudnn and
+                pooling_nd._check_cudnn_acceptable_type(x[0].dtype)):
+            # With cuDNN v3 or greater, use cuDNN implementation for inputs
+            # with spatial dimensions of two or more.
+            if _cudnn_version >= 3000 and self.ndim >= 2:
+                return super(AveragePoolingND, self).backward_gpu(x, gy)
+            # With cuDNN v2, use cuDNN implementation only for inputs with
+            # spatial dimensions of two.
+            elif self.ndim == 2:
+                return super(AveragePoolingND, self).backward_gpu(x, gy)
+
+        n, c = x[0].shape[:2]
+        dims = x[0].shape[2:]
+        ys = gy[0].shape[2:]
+        gx = cuda.cupy.empty_like(x[0])
+        coeff = 1. / functools.reduce(operator.mul, self.ksize)
+
+        in_params, out_params, operation, name = \
+            average_pooling_nd_kernel.AveragePoolingNDKernelBackward.generate(
+                self.ndim)
+        cuda.elementwise(in_params, out_params, operation, name)(
+            gy[0].reduced_view(),
+            *(dims + ys + self.ksize + self.stride + self.pad + (coeff, gx)))
+
+        return gx,
+
+    def create_pool_desc(self):
+        return cudnn.create_pooling_descriptor(
+            self.ksize, self.stride, self.pad,
+            libcudnn.CUDNN_POOLING_AVERAGE_COUNT_INCLUDE_PADDING)
+
+
+def average_pooling_nd(x, ksize, stride=None, pad=0, use_cudnn=True):
+    """N-dimensionally spatial average pooling function.
+
+    This function provides a N-dimensionally generalized version of
+    :func:`~functions.average_pooling_2d`. This acts similarly to
+    :class:`~functions.ConvolutionND`, but it computes the average of input
+    spatial patch for each channel without any parameter instead of computing
+    the inner products.
+
+    Args:
+        x(~chainer.Variable): Input variable.
+        ksize (int or tuple of ints): Size of pooling window. ``ksize=k`` and
+            ``ksize=(k, k, ..., k)`` are equivalent.
+        stride (int or tuple of ints or None): Stride of pooling applications.
+            ``stride=s`` and ``stride=(s,s, ..., s)`` are equivalent. If
```

Comment:
This is a trivial comment, but I think `stride=(s,s, ..., s)` should be `stride=(s, s, ..., s)` (one space is inserted before the second `s`)